### PR TITLE
Stronger check for triggers on columnar tables (#4493).

### PR DESCRIPTION
**Pull request by jeff-davis**
Previously, we used a simple ProcessUtility_hook. Change to use an
object_access_hook instead.

### CODE REVIEW

File: src/backend/columnar/cstore_tableam.c [hunk 1]
```diff
@@ -1143,60 +1127,12 @@ CStoreSubXactCallback(SubXactEvent event, SubTransactionId mySubid,
...
-			createTrigStmt->timing == TRIGGER_TYPE_AFTER)
-		{
-			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-							errmsg(
-								"AFTER ROW triggers are not supported for columnstore access method"),
-							errhint("Consider an AFTER STATEMENT trigger instead.")));
-		}
-	}
-
-	CALL_PREVIOUS_UTILITY();
```
Comment (thanodnl) [inline src/backend/columnar/cstore_tableam.c:1187]: definition of `CALL_PREVIOUS_UTILITY` can be removed as it is not used anymore

File: src/backend/columnar/cstore_tableam.c [hunk 2]
```diff
@@ -1289,6 +1209,78 @@ CStoreTableAMObjectAccessHook(ObjectAccessType access, Oid classId, Oid objectId
...
+}
+
+
+/*
+ * Capture create/drop events and dispatch to the proper action.
+ */
+static void
+CStoreTableAMObjectAccessHook(ObjectAccessType access, Oid classId, Oid objectId, int
+							  subId,
+							  void *arg)
```
Comment (thanodnl) [inline src/backend/columnar/cstore_tableam.c:1265]: nit; We should really figure out how to make our formatter not split between type and variable identifiers, but I would prefer to keep them together like:
```suggestion
CStoreTableAMObjectAccessHook(ObjectAccessType access, Oid classId, Oid objectId,
							  int subId, void *arg)
```

**Review by thanodnl — APPROVED** [review]
small enhancements, changes look good!

File: src/backend/columnar/cstore_tableam.c (hunk 2 above)
Comment (jeff-davis) [reply src/backend/columnar/cstore_tableam.c]: Fixed.

File: src/backend/columnar/cstore_tableam.c (hunk 1 above)
Comment (jeff-davis) [reply src/backend/columnar/cstore_tableam.c:1187]: Fixed.

**Pull request merged**